crates/build/src/lib.rs

```diff
@@ -257,14 +263,31 @@ fn exec_cargo_for_wasm_target(
             args.push("-Zbuild-std-features=panic_immediate_abort".to_owned());
         }
         features.append_to_args(&mut args);
-        let mut env = vec![(
-            "RUSTFLAGS",
-            Some("-C link-arg=-zstack-size=65536 -C link-arg=--import-memory -Clinker-plugin-lto -C target-cpu=mvp"),
-        )];
+        let mut env = Vec::new();
         if rustc_version::version_meta()?.channel == rustc_version::Channel::Stable {
             // Allow nightly features on a stable toolchain
-            env.push(("RUSTC_BOOTSTRAP", Some("1")))
+            env.push(("RUSTC_BOOTSTRAP", Some("1".to_string())))
         }
+        // the linker needs our linker script as file
+        if matches!(target, Target::RiscV) {
+            fs::create_dir_all(&crate_metadata.target_directory)?;
+            let path = crate_metadata
+                .target_directory
+                .join(".riscv_memory_layout.ld");
+            fs::write(&path, include_bytes!("../riscv_memory_layout.ld"))?;
+            env.push((
+                "RUSTFLAGS",
+                Some(format!(
+                    "{} -Clink-arg=-T{}",
+                    target.rustflags(),
```

Comment:
I'd prefer to see the `rustflags()` inline here and below for readability, we are already inside a match on the target.